{"modules": [{"name": "tests.test_api.TestCreateBucketList", "nr_success": 0, "nr_failed": 0, "nr_error": 1, "testcases": [{"name": "test_test", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation \"buckets\" does not exist\n [SQL: '\\nCREATE TABLE items (\\n\\tid SERIAL NOT NULL, \\n\\tname VARCHAR(128) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tdate_modified TIMESTAMP WITHOUT TIME ZONE, \\n\\tdescription VARCHAR(256), \\n\\tdone BOOLEAN, \\n\\tbucket_id INTEGER, \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (name), \\n\\tFOREIGN KEY(bucket_id) REFERENCES buckets (id)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_api.py", "linenr": 9, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}]}, {"name": "tests.test_authentication.TestLogin", "nr_success": 0, "nr_failed": 0, "nr_error": 1, "testcases": [{"name": "test_test", "result": "error", "error": {"message": "sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) duplicate key value violates unique constraint \"pg_type_typname_nsp_index\"\nDETAIL:  Key (typname, typnamespace)=(profiles_id_seq, 2200) already exists.\n [SQL: '\\nCREATE TABLE profiles (\\n\\tid SERIAL NOT NULL, \\n\\thandle VARCHAR(64) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tlast_seen TIMESTAMP WITHOUT TIME ZONE, \\n\\tstatus VARCHAR(140), \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (handle)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_authentication.py", "linenr": 7, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}]}, {"name": "tests.test_models.TestBucket", "nr_success": 0, "nr_failed": 0, "nr_error": 7, "testcases": [{"name": "test_create_Bucket", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation \"buckets\" already exists\n [SQL: '\\nCREATE TABLE buckets (\\n\\tid SERIAL NOT NULL, \\n\\tname VARCHAR(128) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tdate_modified TIMESTAMP WITHOUT TIME ZONE, \\n\\tcreated_by VARCHAR, \\n\\tprofile_id INTEGER, \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (name), \\n\\tFOREIGN KEY(created_by) REFERENCES users (user_name), \\n\\tFOREIGN KEY(profile_id) REFERENCES profiles (id)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 142, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_delete_bucket", "result": "error", "error": {"message": "sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (psycopg2.IntegrityError) duplicate key value violates unique constraint \"users_user_name_key\"\nDETAIL:  Key (user_name)=(crimson) already exists.\n [SQL: 'INSERT INTO users (user_name, password, email, date_created, profile_id) VALUES (%(user_name)s, %(password)s, %(email)s, now(), %(profile_id)s) RETURNING users.id'] [parameters: {'user_name': 'crimson', 'password': 'password', 'email': 'crimson@gmail.com', 'profile_id': None}]", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 601, "function": "run", "line": "testMethod()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 157, "function": "test_delete_bucket", "line": "buckets = bucketlist.Bucket.get_buckets()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/app/models/bucketlist.py", "linenr": 74, "function": "get_buckets", "line": "return Bucket.get_object()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/app/models/bucketlist.py", "linenr": 42, "function": "get_object", "line": "return cls.query.all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2703, "function": "all", "line": "return list(self)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2854, "function": "__iter__", "line": "self.session._autoflush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 1397, "function": "_autoflush", "line": "self.flush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2171, "function": "flush", "line": "self._flush(objects)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2251, "function": "_flush", "line": "subtransactions=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 817, "function": "begin", "line": "nested=nested)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 297, "function": "_begin", "line": "self._assert_active()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 276, "function": "_assert_active", "line": "% self._rollback_exception"}]}}, {"name": "test_query_bucket_by_id", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation \"profiles\" already exists\n [SQL: '\\nCREATE TABLE profiles (\\n\\tid SERIAL NOT NULL, \\n\\thandle VARCHAR(64) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tlast_seen TIMESTAMP WITHOUT TIME ZONE, \\n\\tstatus VARCHAR(140), \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (handle)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 142, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_query_bucket_by_name", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation \"buckets\" already exists\n [SQL: '\\nCREATE TABLE buckets (\\n\\tid SERIAL NOT NULL, \\n\\tname VARCHAR(128) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tdate_modified TIMESTAMP WITHOUT TIME ZONE, \\n\\tcreated_by VARCHAR, \\n\\tprofile_id INTEGER, \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (name), \\n\\tFOREIGN KEY(created_by) REFERENCES users (user_name), \\n\\tFOREIGN KEY(profile_id) REFERENCES profiles (id)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 142, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_query_non_existing_Bucket", "result": "error", "error": {"message": "sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (psycopg2.IntegrityError) duplicate key value violates unique constraint \"users_user_name_key\"\nDETAIL:  Key (user_name)=(crimson) already exists.\n [SQL: 'INSERT INTO users (user_name, password, email, date_created, profile_id) VALUES (%(user_name)s, %(password)s, %(email)s, now(), %(profile_id)s) RETURNING users.id'] [parameters: {'user_name': 'crimson', 'password': 'password', 'email': 'crimson@gmail.com', 'profile_id': None}]", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 601, "function": "run", "line": "testMethod()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 180, "function": "test_query_non_existing_Bucket", "line": "bucket = bucketlist.Bucket.get_bucket(name=\"Doesnotexist bucket\")"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/app/models/bucketlist.py", "linenr": 89, "function": "get_bucket", "line": "return Bucket.get_object(name=name)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/app/models/bucketlist.py", "linenr": 38, "function": "get_object", "line": "return cls.query.filter_by(name=name).first()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2755, "function": "first", "line": "ret = list(self[0:1])"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2547, "function": "__getitem__", "line": "return list(res)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2854, "function": "__iter__", "line": "self.session._autoflush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 1397, "function": "_autoflush", "line": "self.flush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2171, "function": "flush", "line": "self._flush(objects)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2251, "function": "_flush", "line": "subtransactions=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 817, "function": "begin", "line": "nested=nested)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 297, "function": "_begin", "line": "self._assert_active()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 276, "function": "_assert_active", "line": "% self._rollback_exception"}]}}, {"name": "test_query_non_existing_Bucket", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) table \"items\" does not exist\n [SQL: '\\nDROP TABLE items']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 604, "function": "run", "line": "self.tearDown()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 145, "function": "tearDown", "line": "BaseTestCase.tearDown(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 46, "function": "tearDown", "line": "database.drop_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1015, "function": "drop_all", "line": "self._execute_for_all_tables(app, bind, 'drop_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3945, "function": "drop_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 874, "function": "visit_metadata", "line": "table, drop_ok=True, _is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 923, "function": "visit_table", "line": "self.connection.execute(DropTable(table))"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_update_bucket", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation \"users\" does not exist\n [SQL: '\\nCREATE TABLE buckets (\\n\\tid SERIAL NOT NULL, \\n\\tname VARCHAR(128) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tdate_modified TIMESTAMP WITHOUT TIME ZONE, \\n\\tcreated_by VARCHAR, \\n\\tprofile_id INTEGER, \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (name), \\n\\tFOREIGN KEY(created_by) REFERENCES users (user_name), \\n\\tFOREIGN KEY(profile_id) REFERENCES profiles (id)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 142, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}]}, {"name": "tests.test_models.TestItem", "nr_success": 0, "nr_failed": 1, "nr_error": 7, "testcases": [{"name": "test_create_Item", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation \"profiles\" already exists\n [SQL: '\\nCREATE TABLE profiles (\\n\\tid SERIAL NOT NULL, \\n\\thandle VARCHAR(64) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tlast_seen TIMESTAMP WITHOUT TIME ZONE, \\n\\tstatus VARCHAR(140), \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (handle)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 72, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_delete_item", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation \"items\" already exists\n [SQL: '\\nCREATE TABLE items (\\n\\tid SERIAL NOT NULL, \\n\\tname VARCHAR(128) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tdate_modified TIMESTAMP WITHOUT TIME ZONE, \\n\\tdescription VARCHAR(256), \\n\\tdone BOOLEAN, \\n\\tbucket_id INTEGER, \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (name), \\n\\tFOREIGN KEY(bucket_id) REFERENCES buckets (id)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 72, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_query_by_id", "result": "failed", "error": {"message": "AssertionError: False is not true", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 601, "function": "run", "line": "testMethod()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 121, "function": "test_query_by_id", "line": "self.assertTrue(item == query)"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 678, "function": "assertTrue", "line": "raise self.failureException(msg)"}]}}, {"name": "test_query_by_id", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) table \"buckets\" does not exist\n [SQL: '\\nDROP TABLE buckets']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 604, "function": "run", "line": "self.tearDown()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 75, "function": "tearDown", "line": "BaseTestCase.tearDown(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 46, "function": "tearDown", "line": "database.drop_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1015, "function": "drop_all", "line": "self._execute_for_all_tables(app, bind, 'drop_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3945, "function": "drop_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 874, "function": "visit_metadata", "line": "table, drop_ok=True, _is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 923, "function": "visit_table", "line": "self.connection.execute(DropTable(table))"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_query_by_name", "result": "error", "error": {"message": "sqlalchemy.orm.exc.ObjectDeletedError: Instance '<Item at 0x103e106d8>' has been deleted, or its row is otherwise not present.", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 72, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 36, "function": "setUp", "line": "self.new_bucket.items[self.new_item.name] = self.new_item"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/attributes.py", "linenr": 237, "function": "__get__", "line": "return self.impl.get(instance_state(instance), dict_)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/attributes.py", "linenr": 579, "function": "get", "line": "value = state._load_expired(state, passive)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/state.py", "linenr": 592, "function": "_load_expired", "line": "self.manager.deferred_scalar_loader(self, toload)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", "linenr": 703, "function": "load_scalar_attributes", "line": "raise orm_exc.ObjectDeletedError(state)"}]}}, {"name": "test_query_non_existing_item", "result": "error", "error": {"message": "sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (psycopg2.IntegrityError) duplicate key value violates unique constraint \"users_user_name_key\"\nDETAIL:  Key (user_name)=(crimson) already exists.\n [SQL: 'INSERT INTO users (user_name, password, email, date_created, profile_id) VALUES (%(user_name)s, %(password)s, %(email)s, now(), %(profile_id)s) RETURNING users.id'] [parameters: {'user_name': 'crimson', 'password': 'password', 'email': 'crimson@gmail.com', 'profile_id': None}]", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 601, "function": "run", "line": "testMethod()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 113, "function": "test_query_non_existing_item", "line": "item = bucketlist.Item.get_item(name=\"Doesnotexist\")"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/app/models/bucketlist.py", "linenr": 122, "function": "get_item", "line": "return Item.get_object(name=name)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/app/models/bucketlist.py", "linenr": 38, "function": "get_object", "line": "return cls.query.filter_by(name=name).first()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2755, "function": "first", "line": "ret = list(self[0:1])"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2547, "function": "__getitem__", "line": "return list(res)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2854, "function": "__iter__", "line": "self.session._autoflush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 1397, "function": "_autoflush", "line": "self.flush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2171, "function": "flush", "line": "self._flush(objects)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2251, "function": "_flush", "line": "subtransactions=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 817, "function": "begin", "line": "nested=nested)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 297, "function": "_begin", "line": "self._assert_active()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 276, "function": "_assert_active", "line": "% self._rollback_exception"}]}}, {"name": "test_query_non_existing_item", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) table \"buckets\" does not exist\n [SQL: '\\nDROP TABLE buckets']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 604, "function": "run", "line": "self.tearDown()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 75, "function": "tearDown", "line": "BaseTestCase.tearDown(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 46, "function": "tearDown", "line": "database.drop_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1015, "function": "drop_all", "line": "self._execute_for_all_tables(app, bind, 'drop_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3945, "function": "drop_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 874, "function": "visit_metadata", "line": "table, drop_ok=True, _is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 923, "function": "visit_table", "line": "self.connection.execute(DropTable(table))"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_update_item", "result": "error", "error": {"message": "sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) duplicate key value violates unique constraint \"pg_type_typname_nsp_index\"\nDETAIL:  Key (typname, typnamespace)=(profiles_id_seq, 2200) already exists.\n [SQL: '\\nCREATE TABLE profiles (\\n\\tid SERIAL NOT NULL, \\n\\thandle VARCHAR(64) NOT NULL, \\n\\tdate_created TIMESTAMP WITHOUT TIME ZONE, \\n\\tlast_seen TIMESTAMP WITHOUT TIME ZONE, \\n\\tstatus VARCHAR(140), \\n\\tPRIMARY KEY (id), \\n\\tUNIQUE (handle)\\n)\\n\\n']", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 72, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 15, "function": "setUp", "line": "database.create_all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 1007, "function": "create_all", "line": "self._execute_for_all_tables(app, bind, 'create_all')"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", "linenr": 999, "function": "_execute_for_all_tables", "line": "op(bind=self.get_engine(app, bind), **extra)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/schema.py", "linenr": 3918, "function": "create_all", "line": "tables=tables)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1929, "function": "_run_visitor", "line": "conn._run_visitor(visitorcallable, element, **kwargs)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1538, "function": "_run_visitor", "line": "**kwargs).traverse_single(element)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 733, "function": "visit_metadata", "line": "_is_metadata_operation=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/visitors.py", "linenr": 121, "function": "traverse_single", "line": "return meth(obj, **kw)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 767, "function": "visit_table", "line": "include_foreign_key_constraints=include_foreign_key_constraints"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/ddl.py", "linenr": 68, "function": "_execute_on_connection", "line": "return connection._execute_ddl(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1002, "function": "_execute_ddl", "line": "compiled"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}]}, {"name": "tests.test_models.TestProfile", "nr_success": 0, "nr_failed": 0, "nr_error": 4, "testcases": [{"name": "test_create_Profile", "result": "error", "error": {"message": "sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (psycopg2.IntegrityError) duplicate key value violates unique constraint \"users_user_name_key\"\nDETAIL:  Key (user_name)=(crimson) already exists.\n [SQL: 'INSERT INTO users (user_name, password, email, date_created, profile_id) VALUES (%(user_name)s, %(password)s, %(email)s, now(), %(profile_id)s) RETURNING users.id'] [parameters: {'user_name': 'crimson', 'password': 'password', 'email': 'crimson@gmail.com', 'profile_id': None}]", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 601, "function": "run", "line": "testMethod()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 218, "function": "test_create_Profile", "line": "query = user.Profile.get_profile(handle=\"@johndoe_profile\")"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/app/models/user.py", "linenr": 103, "function": "get_profile", "line": "return cls.query.filter_by(handle=handle).first()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2755, "function": "first", "line": "ret = list(self[0:1])"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2547, "function": "__getitem__", "line": "return list(res)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2854, "function": "__iter__", "line": "self.session._autoflush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 1397, "function": "_autoflush", "line": "self.flush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2171, "function": "flush", "line": "self._flush(objects)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2251, "function": "_flush", "line": "subtransactions=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 817, "function": "begin", "line": "nested=nested)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 297, "function": "_begin", "line": "self._assert_active()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 276, "function": "_assert_active", "line": "% self._rollback_exception"}]}}, {"name": "test_delete_Profile", "result": "error", "error": {"message": "sqlalchemy.exc.OperationalError: (psycopg2.extensions.TransactionRollbackError) deadlock detected\nLINE 2: FROM buckets \n             ^\nDETAIL:  Process 78706 waits for AccessShareLock on relation 70882 of database 17086; blocked by process 78709.\nProcess 78709 waits for AccessExclusiveLock on relation 70863 of database 17086; blocked by process 78706.\nHINT:  See server log for query details.\n [SQL: 'SELECT buckets.id AS buckets_id, buckets.name AS buckets_name, buckets.date_created AS buckets_date_created, buckets.date_modified AS buckets_date_modified, buckets.created_by AS buckets_created_by, buckets.profile_id AS buckets_profile_id \\nFROM buckets \\nWHERE buckets.id = %(param_1)s'] [parameters: {'param_1': 1}]", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 209, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 36, "function": "setUp", "line": "self.new_bucket.items[self.new_item.name] = self.new_item"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/attributes.py", "linenr": 237, "function": "__get__", "line": "return self.impl.get(instance_state(instance), dict_)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/attributes.py", "linenr": 584, "function": "get", "line": "value = self.callable_(state, passive)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/strategies.py", "linenr": 557, "function": "_load_for_state", "line": "return self._emit_lazyload(session, state, ident_key, passive)"}, {"filename": "<string>", "linenr": 1, "function": "<lambda>", "line": ""}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/strategies.py", "linenr": 621, "function": "_emit_lazyload", "line": "state, passive=passive)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/strategies.py", "linenr": 490, "function": "_generate_lazy_clause", "line": "state, dict_, ident, passive)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/mapper.py", "linenr": 2564, "function": "_get_state_attr_by_column", "line": "return state.manager[prop.key].impl.get(state, dict_, passive=passive)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/attributes.py", "linenr": 579, "function": "get", "line": "value = state._load_expired(state, passive)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/state.py", "linenr": 592, "function": "_load_expired", "line": "self.manager.deferred_scalar_loader(self, toload)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", "linenr": 698, "function": "load_scalar_attributes", "line": "only_load_props=attribute_names)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", "linenr": 223, "function": "load_on_ident", "line": "return q.one()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2814, "function": "one", "line": "ret = self.one_or_none()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2784, "function": "one_or_none", "line": "ret = list(self)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2855, "function": "__iter__", "line": "return self._execute_and_instances(context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2878, "function": "_execute_and_instances", "line": "result = conn.execute(querycontext.statement, self._params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/elements.py", "linenr": 263, "function": "_execute_on_connection", "line": "return connection._execute_clauseelement(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1053, "function": "_execute_clauseelement", "line": "compiled_sql, distilled_params"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}, {"name": "test_query_by_info", "result": "error", "error": {"message": "sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (psycopg2.InternalError) current transaction is aborted, commands ignored until end of transaction block\n [SQL: 'SELECT users.id AS users_id, users.user_name AS users_user_name, users.password AS users_password, users.email AS users_email, users.date_created AS users_date_created \\nFROM users \\nWHERE users.id = %(param_1)s'] [parameters: {'param_1': 1}]", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 601, "function": "run", "line": "testMethod()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 248, "function": "test_query_by_info", "line": "query = user.Profile.get_profile(handle=\"@new_profile\")"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/app/models/user.py", "linenr": 103, "function": "get_profile", "line": "return cls.query.filter_by(handle=handle).first()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2755, "function": "first", "line": "ret = list(self[0:1])"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2547, "function": "__getitem__", "line": "return list(res)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2854, "function": "__iter__", "line": "self.session._autoflush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 1397, "function": "_autoflush", "line": "self.flush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2171, "function": "flush", "line": "self._flush(objects)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2251, "function": "_flush", "line": "subtransactions=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 817, "function": "begin", "line": "nested=nested)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 297, "function": "_begin", "line": "self._assert_active()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 276, "function": "_assert_active", "line": "% self._rollback_exception"}]}}, {"name": "test_update_info", "result": "error", "error": {"message": "sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) relation \"items\" does not exist\nLINE 2: FROM items \n             ^\n [SQL: 'SELECT items.id AS items_id, items.name AS items_name, items.date_created AS items_date_created, items.date_modified AS items_date_modified, items.description AS items_description, items.done AS items_done, items.bucket_id AS items_bucket_id \\nFROM items \\nWHERE %(param_1)s = items.bucket_id'] [parameters: {'param_1': 1}]", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 597, "function": "run", "line": "self.setUp()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 209, "function": "setUp", "line": "BaseTestCase.setUp(self)"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/base_test_setup.py", "linenr": 36, "function": "setUp", "line": "self.new_bucket.items[self.new_item.name] = self.new_item"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/attributes.py", "linenr": 237, "function": "__get__", "line": "return self.impl.get(instance_state(instance), dict_)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/attributes.py", "linenr": 584, "function": "get", "line": "value = self.callable_(state, passive)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/strategies.py", "linenr": 557, "function": "_load_for_state", "line": "return self._emit_lazyload(session, state, ident_key, passive)"}, {"filename": "<string>", "linenr": 1, "function": "<lambda>", "line": ""}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/strategies.py", "linenr": 635, "function": "_emit_lazyload", "line": "result = q.all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2703, "function": "all", "line": "return list(self)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2855, "function": "__iter__", "line": "return self._execute_and_instances(context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2878, "function": "_execute_and_instances", "line": "result = conn.execute(querycontext.statement, self._params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 945, "function": "execute", "line": "return meth(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/sql/elements.py", "linenr": 263, "function": "_execute_on_connection", "line": "return connection._execute_clauseelement(self, multiparams, params)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1053, "function": "_execute_clauseelement", "line": "compiled_sql, distilled_params"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1189, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1402, "function": "_handle_dbapi_exception", "line": "exc_info"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 203, "function": "raise_from_cause", "line": "reraise(type(exception), exception, tb=exc_tb, cause=cause)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/util/compat.py", "linenr": 186, "function": "reraise", "line": "raise value.with_traceback(tb)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/base.py", "linenr": 1182, "function": "_execute_context", "line": "context)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/engine/default.py", "linenr": 470, "function": "do_execute", "line": "cursor.execute(statement, parameters)"}]}}]}, {"name": "tests.test_models.TestUser", "nr_success": 5, "nr_failed": 0, "nr_error": 1, "testcases": [{"name": "test_create_user", "result": "error", "error": {"message": "sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (psycopg2.InternalError) current transaction is aborted, commands ignored until end of transaction block\n [SQL: 'INSERT INTO users (user_name, password, email, date_created, profile_id) VALUES (%(user_name)s, %(password)s, %(email)s, now(), %(profile_id)s) RETURNING users.id'] [parameters: {'user_name': 'crimson', 'password': 'password', 'email': 'crimson@gmail.com', 'profile_id': None}]", "traceback": [{"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 59, "function": "testPartExecutor", "line": "yield"}, {"filename": "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", "linenr": 601, "function": "run", "line": "testMethod()"}, {"filename": "/Users/georgreen/Projects/CP2A-BucketList-Application/tests/test_models.py", "linenr": 17, "function": "test_create_user", "line": "user_created = user.User.query.all()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2703, "function": "all", "line": "return list(self)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/query.py", "linenr": 2854, "function": "__iter__", "line": "self.session._autoflush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 1397, "function": "_autoflush", "line": "self.flush()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2171, "function": "flush", "line": "self._flush(objects)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 2251, "function": "_flush", "line": "subtransactions=True)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 817, "function": "begin", "line": "nested=nested)"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 297, "function": "_begin", "line": "self._assert_active()"}, {"filename": "/Users/georgreen/.virtualenvs/cp2-flask/lib/python3.6/site-packages/sqlalchemy/orm/session.py", "linenr": 276, "function": "_assert_active", "line": "% self._rollback_exception"}]}}, {"name": "test_delete_user", "result": "success"}, {"name": "test_edit_user", "result": "success"}, {"name": "test_query_by_email", "result": "success"}, {"name": "test_query_by_id", "result": "success"}, {"name": "test_query_non_existing_user", "result": "success"}]}], "metadata": {"command": "/Users/georgreen/.virtualenvs/cp2-flask/bin/nosetests --with-json-extended", "time": 1498155235.114408, "cwd": "/Users/georgreen/Projects/CP2A-BucketList-Application"}}